libs/execution_context/ebpf_core.c

```diff
@@ -1021,19 +1021,115 @@ _ebpf_core_protocol_link_program(
     EBPF_RETURN_RESULT(retval);
 }
 
+static ebpf_result_t
+_ebpf_core_find_matching_link(
+    ebpf_handle_t program_handle,
+    _In_ const ebpf_attach_type_t* attach_type,
+    _In_reads_bytes_(context_data_length) const uint8_t* context_data,
+    size_t context_data_length,
+    _Inout_opt_ ebpf_link_t* previous_link,
+    _Outptr_ ebpf_link_t** link)
```

Comment:
```suggestion
    _Outptr_ ebpf_link_t** link)  noexcept
```

Reply:
Do we need this in C file? There is no precedence in any other functions.

Reply:
oops you're right, that's only for C++ files